src/Products/Signature/Controllers/SignatureApiController.cs

```diff
@@ -1093,27 +1111,26 @@ private void AddSignOptions(string documentType, SignatureOptionsCollection sign
         /// <param name="password">string</param>
         /// <param name="signsCollection">SignatureOptionsCollection</param>
         /// <returns></returns>
-        private SignedDocumentEntity SignDocument(string documentGuid, string password, SignatureOptionsCollection signsCollection)
+        private SignedDocumentEntity SignDocument(string documentGuid, string password, List<SignOptions> signsCollection)
         {
             // set save options
             SaveOptions saveOptions = new SaveOptions();
-            saveOptions.OutputType = OutputType.String;
-            saveOptions.OutputFileName = Path.GetFileName(documentGuid);
             saveOptions.OverwriteExistingFiles = false;
 
-            // set password
-            LoadOptions loadOptions = new LoadOptions();
-            if (!String.IsNullOrEmpty(password))
+            // sign document
+            string tempFilename = Path.GetFileNameWithoutExtension(documentGuid) + "_tmp";
+            string tempPath = Path.Combine(Path.GetDirectoryName(documentGuid), tempFilename + Path.GetExtension(documentGuid));
+            SignedDocumentEntity signedDocument = new SignedDocumentEntity();
+            using (GroupDocs.Signature.Signature signature = new GroupDocs.Signature.Signature(documentGuid, GetLoadOptions(password)))
             {
-                loadOptions.Password = password;
+                SignResult signResult = signature.Sign(tempPath, signsCollection, saveOptions);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'signResult'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Signature-for-.NET-MVC/pullRequest?prid=5325147)